### PR TITLE
[IRGen] Make pointers to accessor functions in layout strings relative

### PR DESCRIPTION
**Pull request by drexin**
rdar://106319336

Pointers embedded in static layout strings should always be relative, so layout strings can reside in read-only memory.

### CODE REVIEW

**Comment by drexin** [comment]
@swift-ci smoke test

**Review by aschwaighofer — APPROVED** [review]

**Comment by drexin** [comment]
@swift-ci smoke test

File: include/swift/ABI/Metadata.h [hunk 1]
```diff
@@ -308,7 +308,7 @@ struct TargetMetadata {
   }
 
   const uint8_t *getLayoutString() const {
-    assert(hasLayoutString());
+    //assert(hasLayoutString());
```
Comment (drexin) [inline include/swift/ABI/Metadata.h:311]: Need to make sure that the flag is properly set for all types now. I'll re-enable this before merging.

**Comment by drexin** [comment]
@swift-ci smoke test

**Comment by drexin** [comment]
@swift-ci smoke test

**Pull request merged**